osu.Game.Benchmarks/BenchmarkBeatmapParsing.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Reflection;
+using BenchmarkDotNet.Attributes;
+using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.Formats;
+using osu.Game.IO;
+using osu.Game.Storyboards;
+
+namespace osu.Game.Benchmarks
+{
+    public class BenchmarkBeatmapParsing : BenchmarkTest
+    {
+        [Benchmark]
+        public Beatmap BenchmarkLegacyBeatmap()
+        {
+            using (var stream = Assembly.GetExecutingAssembly().GetManifestResourceStream(typeof(Program), "Resources.Soleily - Renatus (Gamu) [Insane].osu"))
+            using (var reader = new LineBufferedReader(stream))
+            {
+                var decoder = Decoder.GetDecoder<Beatmap>(reader);
+                return decoder.Decode(reader);
+            }
+        }
+
+        [Benchmark]
+        public Storyboard BenchmarkLegacyStoryboard()
+        {
+            using (var stream = Assembly.GetExecutingAssembly().GetManifestResourceStream(typeof(Program), "Resources.MitiS & MaHi - Blu (Speed Up Ver.) (Ashasaki).osb"))
```

Comment:
We generally don't include full beatmaps/storyboards for copyright reasons. Would prefer if the ones already included in the game/test resources were used instead. Eventually we would construct beatmaps/storyboards on a per-case basis and drill down these benchmarks.

Reply:
Maybe worthwhile checking out the way I did it in the PR i mentioned. It generates a beatmap with enough content to be an intensive benchmark without using existing beatmaps.

Reply:
I'm fine to auto generate beatmaps and storyboards, with several points:
- I need to know specification of legacy beatmap files to generates text that's to be parsed from.
- Patterns coverage.
- A handful entry to run benchmark on ranked maps.

Reply:
1) Create a beatmap and encode it with a `LegacyBeatmapEncoder`. It doesn’t support storyboards right now, though. I’d be fine with omitting it until that’s implemented.
2) Should be made on a per-case basis, like decoding a slider, or decoding a hitcircle, or a hitsound. As above you can just place many of these and encode them to have an “intensive” test.
3) Would love to see something like the beatmap conversion tests, where you can add a new test case based on the file name. That should be enough to then be able to separate out the problematic part and make a custom test for it.